src/test/groovy/org/prebid/server/functional/tests/privacy/GppAmpSpec.groovy

```diff
@@ -0,0 +1,106 @@
+        assert response.ext?.warnings[PREBID]*.code == [999]
+        assert response.ext?.warnings[PREBID]*.message.every { it.contains("GPP string invalid:") }
+    }
+
+    def "PBS should copy consent_string to regs.gpp when consent_string is valid"() {
+        given: "Default amp request with valid consent_string and gpp consent_type"
+        def ampRequest = getGppAmpRequest(gppConsent)
+
+        and: "Save storedRequest into DB"
+        def ampStoredRequest = BidRequest.defaultStoredRequest
+        def storedRequest = StoredRequest.getStoredRequest(ampRequest, ampStoredRequest)
+        storedRequestDao.save(storedRequest)
+
+        when: "PBS processes amp request"
+        defaultPbsService.sendAmpRequest(ampRequest)
+
+        then: "Bidder request should contain regs.gpp from consent_string"
+        def bidderRequest = bidder.getBidderRequest(ampStoredRequest.id)
+        assert bidderRequest.regs.gpp == gppConsent as String
+
+        where:
+        gppConsent << [new TcfEuV2Consent.Builder().build(),
+                       new UspV1Consent.Builder().build()]
+    }
+
+    def "PBS should copy consent_string to user.consent and set gdpr=1 when consent_string is valid and gppSid contains 2"() {
+        given: "Default amp request with valid consent_string and gpp consent_type"
+        def gppConsent = new TcfEuV2Consent.Builder().build()
+        def ampRequest = getGppAmpRequest(gppConsent)
+
+        and: "Save storedRequest into DB"
+        def gppSidIds = [TCF_EU_V2.valueAsInt]
+        def ampStoredRequest = BidRequest.defaultStoredRequest.tap {
+            regs = new Regs(gppSid: gppSidIds, gdpr: null)
+        }
+        def storedRequest = StoredRequest.getStoredRequest(ampRequest, ampStoredRequest)
+        storedRequestDao.save(storedRequest)
+
+        when: "PBS processes amp request"
+        def ampResponse = defaultPbsService.sendAmpRequest(ampRequest)
+
+        then: "PBS shouldn't be perform bidder call"
+        def bidderRequest = bidder.getBidderRequests(ampStoredRequest.id)
+        assert bidderRequest.size() == 0
+
+        and: "Resolved request should contain user.consent, gdpr=1 and gpp form amp stored request"
+        def resolvedRequest = ampResponse.ext.debug.resolvedRequest
+        assert resolvedRequest.user.consent == gppConsent.encodeSection()
+        assert resolvedRequest.regs.gdpr == 1
+        assert resolvedRequest.regs.gppSid == gppSidIds
+    }
+
+    def "PBS should copy consent_string to user.us_privacy when consent_string contains us_privacy and gppSid contains 6"() {
+        given: "Default amp request with valid consent_string and gpp consent_type"
+        def gppConsent = new UspV1Consent.Builder().build()
+        def ampRequest = getGppAmpRequest(gppConsent)
+
+        and: "Save storedRequest into DB"
+        def gppSidIds = [USP_V1.valueAsInt]
+        def ampStoredRequest = BidRequest.defaultStoredRequest.tap {
+            regs = new Regs(gppSid: gppSidIds)
```

Comment:
Same. If we can also set gpp sid through AMP query param, there should be a test for it.

Reply:
Nope